src/Pages/ProductListing/FiltersContainer.jsx

```diff
@@ -0,0 +1,252 @@
+  
+    useEffect(()=>{
+        if(Object.keys(props.categry).length && data?.categories?.length){
+            const indx = data?.categories?.findIndex(el=> el.title === props.categry.categoryName)
+            setCtgrySelected({indx:indx, ctgryId:props.categry.categoryId})
+        }
+    }, [props])
+
+    useEffect(() => {
+        if (response !== null) {
+            setData(response);
+        }
+    }, [response]);
+
+    useEffect(() => {
+        if(data?.categories?.length && ctgrySelected.ctgryId!==0){
+            let fltrArr = data.categories.filter(el=> el.id == ctgrySelected.ctgryId)
+            fltrArr[0].productTags.map((cVal,cIndx) =>{
+                setPrdctTags(prevState => [...prevState, { id:cIndx, prdctTagTile:cVal, isSelected:false} ] )
+            })  
+            fltrArr[0].brandfilters.map((cVal,cIndx) =>{
+                setBrands(prevState => [...prevState, { id:cIndx, brandName:cVal, isSelected:false} ] )
+            })       
+        }            
+    }, [ctgrySelected]);
+
+    useEffect(() => {
+        if (props.clear) {
+            setCtgrySelected({indx:-1, ctgryId:0})
+            setPrdctTags([]);
+            setBrands([])
+        }
+    }, [props]);
+    
+    const handleCategory = (ctgry, cIndx) =>{
+        dispatch({ type: "CATEGORY" ,payload:ctgry.title })
+        setCtgrySelected({indx:cIndx, ctgryId:Number(ctgry.id)})
+        setPrdctTags([]);
+    }
+
+    const handlebrandSearch = (e) =>{
+        setBrandSearchKey(e.target.value);
+        //FUNCTIONALITY yet to be implmented 
+        //Commented for future perspective
+        //return items.filter((item) => {
+            /*
+            // in here we check if our region is equal to our c state
+            // if it's equal to then only return the items that match
+            // if not return All the countries
+            */
+            //    if (item.region == filterParam) {
+            //        return searchParam.some((newItem) => {
+            //          return (
+            //            item[newItem]
+            //                .toString()
+            //                .toLowerCase()
+            //                .indexOf(q.toLowerCase()) > -1
+            //                     );
+            //                 });
+            //             } else if (filterParam == "All") {
+            //                 return searchParam.some((newItem) => {
+            //                     return (
+            //                         item[newItem]
+            //                             .toString()
+            //                             .toLowerCase()
+            //                             .indexOf(q.toLowerCase()) > -1
+            //                     );
+            //                 });
+            //             }
+            //         });
+    }
+
+    const handleProductsTagCheckBox = (prdctTag) =>{
+        let indx = prdctTags.findIndex(el=> el.prdctTagTile === prdctTag)
```

Comment:
Suggestion: can use `const` instead of `let`

Reply:
ok